### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by HKLee2040** [ARM-software/armnn#420]
I follow the instruction and version shown in the following link to build ARMNN
https://github.com/ARM-software/armnn/blob/branches/armnn_20_05/BuildGuideAndroidNDK.md

However, I got several errors message shown below.  Is there anyone who can help?

CXX=aarch64-linux-android29-clang++ CC=aarch64-linux-android21-clang CXX_FLAGS="-fPIE -fPIC" cmake .. -DCMAKE_ANDROID_NDK=$NDK -DCMAKE_SYSTEM_NAME=Android -DCMAKE_SYSTEM_VERSION=29 -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a -DCMAKE_EXE_LINKER_FLAGS="-pie -llog -lz" -DARMCOMPUTE_ROOT=/local/armnn-devenv/ComputeLibrary/ -DARMCOMPUTE_BUILD_DIR=/local/armnn-devenv/ComputeLibrary/build -DBOOST_ROOT=/local/armnn-devenv/boost/install/ -DARMCOMPUTENEON=1 -DARMCOMPUTECL=1 -DARMNNREF=1 -DTF_GENERATED_SOURCES=/local/armnn-devenv/google/tf_pb/ -DBUILD_TF_PARSER=1 -DPROTOBUF_ROOT=/local/armnn-devenv/google/x86_pb_install/
-- Boost version: 1.64.0
-- Found the following Boost libraries:
--   unit_test_framework
--   system
--   filesystem
--   program_options
-- Caffe parser support is disabled
-- Tensorflow Lite parser support is disabled
-- Armnn Serializer support is disabled
-- ArmNN Quantizer support is disabled
-- Including backend common library into the build: /local/armnn-devenv/armnn/src/backends/backendsCommon/common.cmake
-- Including backend common library into the build: /local/armnn-devenv/armnn/src/backends/aclCommon/common.cmake
-- Including backend into the build: /local/armnn-devenv/armnn/src/backends/cl/backend.cmake
-- Including backend into the build: /local/armnn-devenv/armnn/src/backends/neon/backend.cmake
-- Including backend into the build: /local/armnn-devenv/armnn/src/backends/reference/backend.cmake
-- Adding object library dependency to armnn: armnnBackendsCommon
-- Adding object library dependency to armnn: armnnAclCommon
-- Adding object library dependency to armnn: armnnClBackend
-- Adding object library dependency to armnn: armnnClBackendWorkloads
-- Adding object library dependency to armnn: armnnNeonBackend
-- Adding object library dependency to armnn: armnnNeonBackendWorkloads
-- Adding object library dependency to armnn: armnnRefBackend
-- Adding object library dependency to armnn: armnnRefBackendWorkloads
-- Adding object library dependency to UnitTests: armnnBackendsCommonUnitTests
-- Adding object library dependency to UnitTests: armnnAclCommonUnitTests
-- Adding object library dependency to UnitTests: armnnClBackendUnitTests
-- Adding object library dependency to UnitTests: armnnNeonBackendUnitTests
-- Adding object library dependency to UnitTests: armnnRefBackendUnitTests
-- Configuring done
-- Generating done
-- Build files have been written to: /local/armnn-devenv/armnn/build
(base) rogers@atcarch01:/local/armnn-devenv/armnn/build$ make -j16
[  0%] Building CXX object src/backends/aclCommon/CMakeFiles/armnnAclCommon.dir/ArmComputeTensorUtils.cpp.o
[  0%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/TestSharedObject.dir/TestSharedObject.cpp.o
[  1%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefBackend.cpp.o
[  1%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/Filesystem.cpp.o
[  1%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/CommandFileParser.cpp.o
[  1%] Building CXX object src/backends/cl/CMakeFiles/armnnClBackend.dir/ClBackend.cpp.o
[  1%] Building CXX object src/backends/aclCommon/test/CMakeFiles/armnnAclCommonUnitTests.dir/ArmComputeTensorUtilsTests.cpp.o
[  1%] Building CXX object src/backends/backendsCommon/CMakeFiles/armnnBackendsCommon.dir/CpuTensorHandle.cpp.o
[  1%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonBackend.cpp.o
[  2%] Linking CXX shared module testSharedObject/libTestSharedObject.so
[  2%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/Processes.cpp.o
[  2%] Building CXX object src/backends/neon/test/CMakeFiles/armnnNeonBackendUnitTests.dir/NeonCreateWorkloadTests.cpp.o
[  2%] Building CXX object src/backends/cl/test/CMakeFiles/armnnClBackendUnitTests.dir/ClCreateWorkloadTests.cpp.o
[  2%] Built target TestSharedObject
[  2%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/Permute.cpp.o
[  2%] Building CXX object src/backends/reference/test/CMakeFiles/armnnRefBackendUnitTests.dir/ArgMinMaxTests.cpp.o
[  2%] Building CXX object src/backends/cl/test/CMakeFiles/armnnClBackendUnitTests.dir/ClEndToEndTests.cpp.o
[  2%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/ArgMinMax.cpp.o
[  2%] Building CXX object src/backends/cl/workloads/CMakeFiles/armnnClBackendWorkloads.dir/ClAbsWorkload.cpp.o
[  2%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonAbsWorkload.cpp.o
[  3%] Building CXX object src/backends/backendsCommon/CMakeFiles/armnnBackendsCommon.dir/DynamicBackend.cpp.o
[  3%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/DataLayoutIndexed.cpp.o
[  3%] Building CXX object src/backends/aclCommon/CMakeFiles/armnnAclCommon.dir/BaseMemoryManager.cpp.o
[  3%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/BackendIdTests.cpp.o
[  3%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/CommandLineProcessor.cpp.o
[  3%] Building CXX object src/backends/aclCommon/test/CMakeFiles/armnnAclCommonUnitTests.dir/MemCopyTests.cpp.o
[  3%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/Activation.cpp.o
[  4%] Building CXX object src/backends/reference/test/CMakeFiles/armnnRefBackendUnitTests.dir/RefCreateWorkloadTests.cpp.o
[  5%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/DotSerializer.cpp.o
[  5%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonInterceptorScheduler.cpp.o
[  5%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/BatchNormImpl.cpp.o
[  5%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefTensorHandle.cpp.o
[  6%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/BackendProfilingTests.cpp.o
[  6%] Built target armnnAclCommon
[  6%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/BackendRegistryTests.cpp.o
[  6%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonActivationWorkload.cpp.o
[  7%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/GatordMockService.cpp.o
[  7%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonLayerSupport.cpp.o
[  7%] Building CXX object src/backends/cl/CMakeFiles/armnnClBackend.dir/ClBackendContext.cpp.o
[  7%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefLayerSupport.cpp.o
[  7%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/BatchToSpaceNd.cpp.o
[  8%] Building CXX object src/backends/cl/workloads/CMakeFiles/armnnClBackendWorkloads.dir/ClActivationWorkload.cpp.o
[  8%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/HeapProfiling.cpp.o
[  8%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/LeakChecking.cpp.o
[  8%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/ModelAccuracyChecker.cpp.o
[  8%] Building CXX object src/backends/backendsCommon/CMakeFiles/armnnBackendsCommon.dir/DynamicBackendUtils.cpp.o
[  8%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/PeriodicCounterCaptureCommandHandler.cpp.o
[  8%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefMemoryManager.cpp.o
[  8%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefRegistryInitializer.cpp.o
[  9%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/Broadcast.cpp.o
[  9%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonAdditionWorkload.cpp.o
[  9%] Building CXX object src/backends/cl/CMakeFiles/armnnClBackend.dir/ClContextControl.cpp.o
[  9%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonArgMinMaxWorkload.cpp.o
[  9%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonRegistryInitializer.cpp.o
[  9%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/PeriodicCounterSelectionResponseHandler.cpp.o
[  9%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/CsvReader.cpp.o
[  9%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/ConvImpl.cpp.o
[  9%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/gatordmock/StreamMetadataCommandHandler.cpp.o
[  9%] Built target armnnAclCommonUnitTests
[  9%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/Debug.cpp.o
[  9%] Building CXX object src/backends/cl/workloads/CMakeFiles/armnnClBackendWorkloads.dir/ClAdditionWorkload.cpp.o
[  9%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/CommonTestUtils.cpp.o
[  9%] Building CXX object src/backends/reference/CMakeFiles/armnnRefBackend.dir/RefWorkloadFactory.cpp.o
[  9%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/DynamicBackendTests.cpp.o
[  9%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonTensorHandleFactory.cpp.o
[  9%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/FloatingPointConverter.cpp.o
[  9%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/DepthToSpace.cpp.o
[  9%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/timelineDecoder/TimelineCaptureCommandHandler.cpp.o
[ 10%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonBatchNormalizationWorkload.cpp.o
[ 10%] Building CXX object src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonBatchToSpaceNdWorkload.cpp.o
[ 10%] Building CXX object src/backends/cl/CMakeFiles/armnnClBackend.dir/ClLayerSupport.cpp.o
[ 10%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/VerificationHelpers.cpp.o
[ 10%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/DetectionPostProcess.cpp.o
[ 10%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/timelineDecoder/TimelineDecoder.cpp.o
[ 10%] Building CXX object CMakeFiles/gatordMockService.dir/tests/profiling/timelineDecoder/TimelineDirectoryCaptureCommandHandler.cpp.o
[ 11%] Building CXX object CMakeFiles/armnnUtils.dir/src/armnnUtils/ParserHelper.cpp.o
[ 11%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonTimer.cpp.o
[ 11%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/Dequantize.cpp.o
[ 12%] Building CXX object src/backends/neon/CMakeFiles/armnnNeonBackend.dir/NeonWorkloadFactory.cpp.o
[ 13%] Linking CXX static library libgatordMockService.a
Error running link command: No such file or directory
CMakeFiles/gatordMockService.dir/build.make:302: recipe for target 'libgatordMockService.a' failed
make[2]: *** [libgatordMockService.a] Error 2
CMakeFiles/Makefile2:189: recipe for target 'CMakeFiles/gatordMockService.dir/all' failed
make[1]: *** [CMakeFiles/gatordMockService.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

...

[ 16%] Linking CXX static library libarmnnUtils.a
Error running link command: No such file or directory
CMakeFiles/armnnUtils.dir/build.make:458: recipe for target 'libarmnnUtils.a' failed
make[2]: *** [libarmnnUtils.a] Error 2
CMakeFiles/Makefile2:226: recipe for target 'CMakeFiles/armnnUtils.dir/all' failed
make[1]: *** [CMakeFiles/armnnUtils.dir/all] Error 2
...

[ 36%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/InstanceNormalizationTestImpl.cpp.o
/local/armnn-devenv/armnn/src/backends/neon/workloads/NeonSoftmaxBaseWorkload.cpp:24:100: error: implicit conversion changes signedness: 'unsigned int' to 'int32_t' (aka 'int')
      [-Werror,-Wsign-conversion]
    return arm_compute::NESoftmaxLayer::validate(&aclInputInfo, &aclOutputInfo, descriptor.m_Beta, aclAxis);
           ~~~~~~~~~~~                                                                             ^~~~~~~
[ 36%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/L2NormalizationTestImpl.cpp.o
[ 37%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/LogSoftmaxTestImpl.cpp.o
1 error generated.
[ 37%] Building CXX object src/backends/cl/workloads/CMakeFiles/armnnClBackendWorkloads.dir/ClSoftmaxBaseWorkload.cpp.o
src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/build.make:950: recipe for target 'src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonSoftmaxBaseWorkload.cpp.o' failed
make[2]: *** [src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/NeonSoftmaxBaseWorkload.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 37%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/LstmTestImpl.cpp.o
[ 37%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/MaximumTestImpl.cpp.o
[ 37%] Building CXX object src/backends/backendsCommon/test/CMakeFiles/armnnBackendsCommonUnitTests.dir/layerTests/MinimumTestImpl.cpp.o
[ 38%] Building CXX object src/backends/cl/workloads/CMakeFiles/armnnClBackendWorkloads.dir/ClSoftmaxFloatWorkload.cpp.o
[ 38%] Building CXX object src/backends/reference/workloads/CMakeFiles/armnnRefBackendWorkloads.dir/RefPreluWorkload.cpp.o
CMakeFiles/Makefile2:1584: recipe for target 'src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/all' failed
make[1]: *** [src/backends/neon/workloads/CMakeFiles/armnnNeonBackendWorkloads.dir/all] Error 2

**Comment by james-conroy-arm** [comment]
Hi @HKLee2040 ,

Which revision/branch of ArmNN and Compute Library are you using?

Cheers,
James

**Comment by HKLee2040** [comment]
ComputeLibrary:
git describe --tags
v20.5

ArmNN:
git checkout branches/armnn_20_05
git pull

**Comment by FrancisMurtagh-arm** [comment]
Closing for now due to age of ticket.

However @HKLee2040 if you are still having issues building please reopen and I can try to help.

Francis.
